mergify_engine/engine/commands_runner.py

```diff
@@ -176,7 +176,10 @@ def log(comment_out: str, result: typing.Optional[check_api.Result] = None) -> N
     if (
         user
         and user["id"] != config.BOT_USER_ID
-        and not await ctxt.repository.has_write_permission(user)
+        and not (
+            await ctxt.repository.has_write_permission(user)
+            or user["id"] == ctxt.pull["user"]["id"]
+        )
```

Comment:
This is barely readable now, I'd reverse the `and` in `or`.
https://fr.wikipedia.org/wiki/Alg%C3%A8bre_de_Boole_(logique)

following https://fr.wikipedia.org/wiki/Lois_de_De_Morgan law
```python
       user
        and user["id"] != config.BOT_USER_ID
        and not (
            await ctxt.repository.has_write_permission(user)
            or user["id"] == ctxt.pull["user"]["id"]
        )
```
can be rewritten
```python
        not (
          user is not None
          or user["id"] == config.BOT_USER_ID
          or (await ctxt.repository.has_write_permission(user)
              or user["id"] == ctxt.pull["user"]["id"])
         )
```

since you only have `or` now you can remove the `()` which makes it simpler:
```python
        not (
          user is not None
          or user["id"] == config.BOT_USER_ID
          or await ctxt.repository.has_write_permission(user)
          or user["id"] == ctxt.pull["user"]["id"]
         )
```

now you have a list of conditions that makes it possible to run a command, with `or` around them, and just a `not` in front of those to negate the condition (and post the comment to say it's disallowed).